README.md

```diff
@@ -77,25 +78,84 @@ Django styleguide that we use in [HackSoft](https://hacksoft.io).
 
 ## Overview
 
+The core of the Django Styleguide can be summarized as follows:
+
 **In Django, business logic should live in:**
 
-* Model properties (with some exceptions).
-* Model `clean` method for additional validations (with some exceptions).
 * Services - functions, that mostly take care of writing things to the database.
 * Selectors - functions, that mostly take care of fetching things from the database.
+* Model properties (with some exceptions).
+* Model `clean` method for additional validations (with some exceptions).
 
 **In Django, business logic should not live in:**
 
 * APIs and Views.
 * Serializers and Forms.
 * Form tags.
 * Model `save` method.
+* Custom managers or querysets.
+* Signals.
 
 **Model properties vs selectors:**
 
 * If the property spans multiple relations, it should better be a selector.
 * If the property is non-trivial & can easily cause `N + 1` queries problem, when serialized, it should better be a selector.
 
+The general idea is to "separate concerns" so those concerns can be maintainable / testable.
+
+## Why not?
+
+> 🤔 Why not put your business logic in APIs / Views / Serializers / Forms?
+
+Relying on generic APIs / Views, with the combination of serializers & forms does 2 major things:
+
+1. Fragments the business logic in multiple places, making it really hard to trace the data flow.
+2. Hides things from you. In order to change something, you need to know the inner-workings of the abstraction that you are using.
+
+Generic APIs & Views, in combination with serializers & forms, is really great for the straightforward "CRUD for a model" case.
+
+From our experience, so far, this straightforward case rarely happens. **And once you leave the happy CRUD path, things start to get messy.**
+
+And once things start to get messy, you need more "boxes", to organize your code in a better way.
+
+This styleguide aims to:
+
+1. Give you those "boxes".
+1. Help you figure out your own "boxes", for your own specific context & needs.
+
+---
+
+> 🤔 Why not put your business logic in custom managers and/or querysets?
+
+This is actually a good idea & you might introduce custom managers & querysets, that can expose better API, tailored to your domain.
+
+But trying to place all of your business logic in a custom manager is not a great idea, because of the following:
+
+1. Business logic has it's own domain, which is not always directly mapped to your data model (models)
+1. Business logic most often spans across multiple models, so it's really hard to choose where to place something.
+    - Lets say you have a custom piece of logic that touches models `A`, `B`, `C` and `D`. Where do you put it?
+1. There can be additional calls to 3rd party systems. You don't want those in your custom manager methods.
```

Comment:
```suggestion
This is actually a good idea & you might introduce custom managers & querysets, that can expose better API, tailored to your domain.

But trying to place all of your business logic in a custom manager is not a great idea, because of the following:

1. Business logic has its own domain, which is not always directly mapped to your data model (models)
1. Business logic most often spans across multiple models, so it's really hard to choose where to place something.
    - Let's say you have a custom piece of logic that touches models `A`, `B`, `C`, and `D`. Where do you put it?
1. There can be additional calls to 3rd party systems. You don't want those in your custom manager methods.
```